### DISCUSSION TITLE
Podcast Playlists

### DISCUSSION THREAD

**Discussion by billdeys** [advplyr/audiobookshelf#986] [Ideas]
Just found aduiobookshelf and I like it. I listen to a lot of podcasts and I'm a pack rat. 

I've been looking for a self hosted platform so I can keep podcasts on my home NAS but also listen on my phone and computers. So far I like audiobookshelf except for a few missing features that are almost deal breakers. 

I need a playlist feature, maybe collections can be used with some tweaks. Mainly, a smart playlist that can look at a number of podcasts (or all) and make a playlist of the unplayed episodes and remove them as they are played. 

Also would be nice to be able to sync downloads to my mobile devices when I'm local for when I'm not to save data, without having to select individual episodes. 

This is great for archiving and re-listening to old single episodes, love to see a few more features to be able to use it full time!

Thank you.

**Comment by advplyr** [comment]
I was thinking we would add another page for podcasts that would be a feed of recent unfinished episodes. I think this is how most podcast apps do it.
Do you have an example of an app that has this smart playlist that you mention?

**Reply by billdeys** [reply]
Most podcast apps aside from Apple Podcasts have a feature similar. The app I use most is Overcast.

**Reply by TunaLobster** [reply]
Podcast Republic has a lot of detailed playlist controls.

**Comment by Strit** [comment]
I would also like a "smart queue" for podcast episodes. One where new episodes are added automatically and removed from the playlist when they are marked as finished. Maybe even an option to remove downloaded episodes if no user has it in a queue.

**Comment by ehula** [comment]
Yes. Smart Playlists would be incredible.

- Each Smart Playlist would auto-populate with new episodes from selected podcasts.
- Allow sorting. Default: oldest episode first.
- Allow episodes to be manually sorted. New episodes are manually added to the end of playlist.
- Allow episodes to be manually removed from the playlist.
- Played episodes are removed from the playlist.

Separately, it would be great to have a per-podcast setting to delete from media played episodes after a set time.

**Comment by idojius86** [comment]
I would also find it very interesting to have the option  have the option of adding multiple podcast episodes to a playlist, instead of adding them one by one. 

And, as mentioned above, It would  be nice to have a sorting system, by date, or lenght.

**Comment by shaftspanner** [comment]
For me, playlists of podcasts that automatically start one episode after another are a potentially life-saving feature.  I regularly make car journeys of 1.5 to 2 hours which means multiple podcasts - without this feature I have to interact with my phone to select a new podcast.
As for what I'd like to see, I'm fairly easy and would be prepared to use more or less whatever is presented.  Personally I'd be happy if I could just hit play on the existing list of latest podcasts and it played them from oldest to newest - anything more than that would be a bonus.

As an aside, Audiobookshelf is my most  used bit of selfhosted software - if only I had the skills to contribute more to the project.

**Reply by Abateman121** [reply]
I would love a smart playlist feature for podcast like overcast does.

**Reply by mogster85** [reply]
Having to manually play each item in a playlist is my biggest issue with the mobile app by far, also due to having to interact with the app while driving. It's absolutely wonderful other than that.

**Comment by linuxdaemon** [comment]
Definitely agree on smart playlists. I personally get a lot of use out of the one in podcast addict, using it to sort all unplayed episodes of all podcasts by release date and listen from oldest to newest. This could even just be a diffetent UI for the podcast library page, rather than grouping by podcast, show all episodes. Then use the same filtering and sorting interface potentially

**Comment by MasterChiefmas** [comment]
Until it's integrated, I'm working on a Powershell script to build a playlist for a podcast, although I've run into a snag figuring out how to get the list episodes of the podcast. I can post it once I get it done. It'll be basic at first, but I may keep refining it until something shows up built in to ABS.

**Comment by TunaLobster** [comment]
I've created this to control my playlists. It works well enough for me and is fairly simple. https://github.com/TunaLobster/PyAutoABSPlaylist
